Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/distance.h

```diff
@@ -116,67 +128,422 @@ sample_triangles(const TriangleRange& triangles, double distance, OutputIterator
   return out;
 }
 
-template <class Kernel, class TriangleRange1, class TriangleRange2>
+#ifdef CGAL_LINKED_WITH_TBB
+template <class AABB_tree, class Point_3>
+struct Distance_computation{
+  const AABB_tree& tree;
+  const std::vector<Point_3>& sample_points;
+  Point_3 initial_hint;
+  cpp11::atomic<double>* distance;
+
+  Distance_computation(const AABB_tree& tree, const Point_3 p, const std::vector<Point_3>& sample_points, cpp11::atomic<double>* d)
+    : tree(tree)
+    , sample_points(sample_points)
+    , distance(d)
+    , initial_hint(p)
+  {}
+
+  void
+  operator()(const tbb::blocked_range<std::size_t>& range) const
+  {
+    Point_3 hint = initial_hint;
+    double hdist = 0;
+    for( std::size_t i = range.begin(); i != range.end(); ++i)
+    {
+      double d = CGAL::sqrt( squared_distance(hint,sample_points[i]) );
+      if (d>hdist) hdist=d;
+    }
+
+    if (hdist > distance->load(cpp11::memory_order_acquire))
+      distance->store(hdist, cpp11::memory_order_release);
+  }
+};
+#endif
+
+/**
+ * @brief enum used to select the sampling method in the function `sample_triangle_mesh`
+ */
+enum Sampling_method{
+  RANDOM_UNIFORM =0,
+  GRID,
+  MONTE_CARLO
+};
+/** fills `sampled_points` with points taken on the mesh in a manner depending on `method`.
+ * @tparam TriangleMesh a model of the concept `FaceListGraph`.
+ * @param m the triangle mesh that will be sampled
+ * @param parameter depends on `method` :
+ *                  RANDOM_UNIFORM and MONTE_CARLO: the number of points per squared area unit.
+ *                  GRID : The distance between two consecutive points in the grid.
+ *
+ * @param np a sequence of \ref namedparameters for `tm` among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *    \cgalParamBegin{vertex_point_map} the property map with the points associated to the vertices of `pmesh` \cgalParamEnd
+ *    \cgalParamBegin{geom_traits} an instance of a geometric traits class, model of `Kernel`\cgalParamEnd
+ * \cgalNamedParamsEnd
+ *
+ * @param method defines the method of sampling.
+ *
+ * @tparam Sampling_method defines the method of sampling.
+ *                         Possible values are :
+ *- `RANDOM_UNIFORM` : points are generated in a random and uniform way, depending on the area of each triangle.,
+ *
+ *- `GRID` : points are generated in a grid, with a minimum of one point per triangle.
+ *
+ *- `MONTE_CARLO` : points are generated randomly in each triangle. Their number in each triangle is proportional to the corresponding face area with a minimum
+ *                  of 1.
+ *
+ */
+template<class TriangleMesh, class NamedParameters>
+void sample_triangle_mesh(const TriangleMesh& m,
+                          double parameter,
+                          std::vector<typename GetGeomTraits<TriangleMesh,
```

Comment:
`GetGeomTraits` cannot appear in the doc. Add `Point_3` as template parameter documenting that is should be the point of the Geom traits in the named parameters